tests/Functional/CatalogPromotionRemovalAnnouncerTest.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Tests\Functional;
+
+use Fidry\AliceDataFixtures\LoaderInterface;
+use Fidry\AliceDataFixtures\Persistence\PurgeMode;
+use Sylius\Bundle\CoreBundle\CatalogPromotion\Announcer\CatalogPromotionRemovalAnnouncerInterface;
+use Sylius\Component\Core\Model\CatalogPromotion;
+use Sylius\Component\Core\Model\CatalogPromotionInterface;
+
+final class CatalogPromotionRemovalAnnouncerTest extends AbstractWebTestCase
+{
+    /** @test */
+    public function it_puts_catalog_promotion_into_processing_state(): void
+    {
+        $this->createClient(['test_case' => 'CatalogPromotionProcessingState']);
```

Comment:
Oh, it's redundant in this case as we can check the last state of CP even if the CP is deleted.

Answering your question, I introduced it some time ago ([here](https://github.com/Sylius/Sylius/pull/14290/files#diff-6805ec64cd2f007f6754ab7fe0ad41828ca3f0a7ea888f494eb8d5f3f623536eR31-R38)) just to test the intermediate `processing` test because in tests we use sync processing only.